cluster-autoscaler/proposals/provisioning-request.md

```diff
@@ -0,0 +1,289 @@
+# Provisioning Request CRD
+
+author: kisieland
+
+## Background
+
+Currently CA does not provide any way to express that a group of pods would like
+to have a capacity available.  
+This is caused by the fact that each CA loop picks a group of unschedulable pods
+and works on provisioning capacity for them, meaning that the grouping is random
+(as it depends on the kube-scheduler and CA loop interactions).  
+This is especially problematic in couple of cases:
+
+  - Users would like to have all-or-nothing semantics for their workloads.
```

Comment:
It cannot be guaranteed to succeed, but what we propose in the `kubernetes.io.atomic-scale-up` mode is to avoid paying for the partially provisioned VMs and clean them up, see  the `kubernetes.io.atomic-scale-up class` section